SU2_CFD/src/iteration/CFluidIteration.cpp

```diff
@@ -124,7 +124,7 @@ void CFluidIteration::Iterate(COutput* output, CIntegration**** integration, CGe
 
   if (config[val_iZone]->GetKind_Scalar_Model() != NO_SCALAR_MODEL){
     config[val_iZone]->SetGlobalParam(RANS, RUNTIME_SCALAR_SYS);
-    integration[val_iZone][val_iInst][SCALAR_SOL]->MultiGrid_Iteration(geometry, solver, numerics, config, 
+    integration[val_iZone][val_iInst][SCALAR_SOL]->SingleGrid_Iteration(geometry, solver, numerics, config, 
```

Comment:
I already wondered why MultigridIteration for Scalar. This does make a difference, e.g. Preprocessing is also called after the iteration for Multigrid! I think in this case  the restarts (for Marks work) should be fine but I am not entirely sure for the flamelet cases. If there are potential Testcases with table lookup I highly recommend adding those here. I already put the structure in place. Now that the primal restart is working I would go for a restarted case!

Reply:
I feel that when we have confidence in multigrid, we should activate it for the additional solvers like turbulence and scalars. But at this moment I do not see a large benefit in using multigrid because it is not robust.